cardano-node/src/Cardano/Tracing/Tracers.hs

```diff
@@ -154,11 +154,21 @@ instance ElidingTracer
   doelide (WithSeverity _ (WithTip _ (ChainDB.TraceGCEvent _))) = True
   doelide _ = False
   conteliding _tform _tverb _tr _ (Nothing, _count) = return (Nothing, 0)
-  conteliding _tform _tverb tr ev (_old, count) = do
-      when (count > 0 && count `mod` 100 == 0) $ do  -- report every 100th elided message
+  conteliding _tform _tverb tr ev@(WithSeverity _ (WithTip _ (ChainDB.TraceGCEvent _))) (_old, count) = do
+      when (count > 0 && count `mod` 100 == 0) $ do  -- report every 100th message
           meta <- mkLOMeta (getSeverityAnnotation ev) (getPrivacyAnnotation ev)
           traceNamedObject tr (meta, LogValue "messages elided so far" (PureI $ toInteger count))
       return (Just ev, count + 1)
+  conteliding _tform _tverb tr ev@(WithSeverity _ (WithTip _ (ChainDB.TraceLedgerReplayEvent (LedgerDB.ReplayedBlock pt replayTo)))) (_old, count) = do
+      let slotno = toInteger $ unSlotNo (realPointSlot pt)
+          endslot = toInteger $ withOrigin 0 unSlotNo (pointSlot replayTo)
+          progress :: Double = (fromInteger slotno * 100.0) / fromInteger (max slotno endslot)
```

Comment:
When the replay starts from scratch, the progress will go from 0-100 %, good. But if the replay starts from a later slot, then it will go from X-100 %, which could be a bit misleading? On each startup, we'll do a small replay of the last `k` blocks, so this is actually very common.

For example: we start replaying from slot 3,000,000 up to slot 3,002,160, the progress will be: 99.9% -> 100%. The more slots the chain gets, the less meaningful the progress will be.

The solution is to figure out the starting point of the replay. You can use the first `ReplayedBlock` message for this. Alternatively, before replay starts, either `ReplayFromGenesis` or `ReplayFromSnapshot` is traced, so you could also use those as the starting point.

Reply:
what do you want to appear in the logs? shall we print the range of the replayed blocks for the first message?

Reply:
No, I mean use the first message (or `ReplayFrom*`) to compute the correct progress for the following messages.

For example, when replaying from slot 3,000,000 up to slot 3,002,160, the first message (or `ReplayFromSnapshot`) will mention slot 3,000,000. So you can compute the progress as (something like): `(slotNo - startSlotNo) / endSlotNo`. This means you'll have to store some state, i.e., the start slot.